Realm/RLMObjectSchema.mm

```diff
@@ -267,6 +278,12 @@ + (NSArray *)propertiesForClass:(Class)objectClass isSwift:(bool)isSwiftClass {
             }
             if (auto type = RLMCoerceToNil(propertyType)) {
                 if (existing == NSNotFound) {
+                    // Check to see if this optional property is an underlying storage property for a Swift lazy var.
+                    // Managed lazy vars are't allowed.
+                    // NOTE: Revisit this once property behaviors are implemented in Swift.
+                    if ([self _isDisallowedLazyProperty:propertyName ignoredProps:ignoredProperties]) {
+                        @throw RLMException(@"Lazy managed properties are not allowed on Realm Swift object classes.");
```

Comment:
The exception message could include a reference to the property name and a hint that this needs to be either ignored or not defined as lazy.